nibabel/streamlines/trk.py

```diff
@@ -559,8 +559,8 @@ def _read_header(fileobj):
 
             # Read the header in one block.
             header_str = f.read(header_2_dtype.itemsize)
-            header_rec = np.fromstring(string=header_str, dtype=header_2_dtype)
-
+            header_rec = np.frombuffer(buffer=bytearray(header_str),
```

Comment:
This still makes a copy from `header_str` to `bytearray(header_str)`. Instead, we can create a bytearray and read directly into it:

```Python
            header_buf = bytearray(header_2_dtype.itemsize)
            n_read = f.readinto(header_buf)
            # Maybe if n_read != header_2_dtype.itemsize: raise HeaderError(...)
            header_rec = np.frombuffer(buffer=header_buf, dtype=header_2_dtype)
```